conmon/conmon.c

```diff
@@ -1333,6 +1361,9 @@ int main(int argc, char *argv[])
 		exit(127);
 	}
 
+	if (opt_exit_command)
```

Comment:
Can we move this down to around line 1494, inside the `if(!opt_exec)`, so we have the exit code, and pass that along as one of the arguments to the exit command? That way we can guarantee the container process has exited and it's safe to do cleanup routines (like unmount container storage and delete network namespaces), and capture the exit code at the same time?

Reply:
the atexit registers a callback that is called by `exit(3)`, so this will really be the last thing done by conmon before it exits

Reply:
Yeah, that would work fine for our purposes. We probably don't want this for OPT_EXEC right now but we can avoid that by just not adding the flag in such cases

Reply:
it will be done in any case, also for `exec`.

Reply:
Probably fine, we can work around that on our end

Reply:
Why are we registering this `atexit` and then launching the process with `g_spawn_sync`?  Can't we just [`execvp`][1] the exit program after we finish our cleanup?  Then we wouldn't have the `conmon` process idling while the exit program was working, and we'd save some new-process setup costs.

[1]: http://pubs.opengroup.org/onlinepubs/9699919799.2016edition/functions/execvp.html